src/main/java/net/sf/jabref/logic/exporter/BibDatabaseWriter.java

```diff
@@ -158,7 +158,11 @@ public E savePartOfDatabase(BibDatabaseContext bibDatabaseContext,
 
         session = saveSessionFactory.createSaveSession(preferences.getEncodingOrDefault(), preferences.getMakeBackup());
 
-        writeDatabaseID(bibDatabaseContext.getDatabase().getDatabaseID());
+        Optional<String> sharedDatabaseIDOptional = bibDatabaseContext.getDatabase().getSharedDatabaseID();
+
+        if (sharedDatabaseIDOptional.isPresent()) {
```

Comment:
you could use orElseThrow() (http://stackoverflow.com/questions/28596790/throw-an-exception-if-an-optional-is-present). But from my side it's ok how it is.